### PR TITLE
Addressing Issues 1076 and 1080

### PR DESCRIPTION
**Pull request by SteveGilham**
This is the one about the lack of CodeTaskFactory for `dotnet build`, following the related thread for [MSBuild issue #616](https://github.com/Microsoft/msbuild/issues/616); so if the Roslyn version of the assembly is not present, a warning is emitted, otherwise use it to generate the task.  Manual testing (by patching the nuget package) allowed an application build and launch using `dotnet` (see [comment](https://github.com/picoe/Eto/issues/1080#issuecomment-388638019) on issue #1080)

### CODE REVIEW

**Comment by SteveGilham** [comment]
Still TODO would be making this transparent to existing consumers, but adding the RoslynCodeTaskfactory package as a dependency of the eto.platform.mac64.nupkg, and referencing the former directly from the latter (which may be tricky).

**Comment by SteveGilham** [comment]
That does leave the version number in 3 places, though if it's sufficient for purpose there's not great rush to upgrade.

**Comment by cwensley** [comment]
Hey @SteveGilham, thanks a ton for the PR!  This is looking really great!

I'm thinking a better way to resolve this would be to include a compiled version of the tasks.. that way we wouldn't need any extra dependencies.  However, I haven't looked into it yet, but I think it may need separate compiled task assemblies for .net core and for full .net.  Do you know anything about this?

**Comment by cra0zy** [comment]
> I'm thinking a better way to resolve this would be to include a compiled version of the tasks.. that way we wouldn't need any extra dependencies. However, I haven't looked into it yet, but I think it may need separate compiled task assemblies for .net core and for full .net. Do you know anything about this?

I know about this, and yes, you need separate versions compiled, see: https://github.com/cra0zy/MonoGame/tree/mnu/Tools/MonoGame.Content.Builder if you wanna see it in action.

**Comment by SteveGilham** [comment]
Indeed -- even if the code is the same, the linkage information depends on the runtime.  Consuming the Roslyn-based package simply hides that selection.

Having worked in the past of projects that built the build tasks they then consumed, I can say that the build/test/repeat cycle can be a bit painful when the task dlls get locked in processes you don't want to have to kill.

File: build/MacTemplate/MacTemplate.targets [hunk 1]
```diff
@@ -12,10 +12,12 @@
 		<OutputAppPath Condition="'$(OutputAppPath)'==''">$(TargetDir)\$(MacBundleName).app</OutputAppPath>
 		<MacSetDefaultRunConfiguration Condition="'$(MacSetDefaultTarget)' == ''">True</MacSetDefaultRunConfiguration>
 		<MacBuildBundle Condition="$(MacBuildBundle) == '' and $(ProjectTypeGuids) == ''">True</MacBuildBundle>
+		<RoslynCodeTaskProps>$(MSBuildThisFileDirectory)..\..\..\roslyncodetaskfactory\2.0.7\Sdk\Sdk.props</RoslynCodeTaskProps>
```
Comment (cwensley) [inline build/MacTemplate/MacTemplate.targets]: Hm, this is the only line I really have a problem with.  Is this (and the Import below) needed if the nuget package is installed for the project?

**Comment by cwensley** [comment]
Thanks @SteveGilham and @cra0zy for the added information.

I agree that going with the `RoslynCodeTaskFactory` is a good solution at this point and will reduce the overall maintenance of going with a compiled task library.  This PR looks good except the one line that I commented on.  For example, it wouldn't work if the project referenced a different version than 2.0.7.

**Comment by SteveGilham** [comment]
There is that problem, [as I noted earlier](https://github.com/picoe/Eto/pull/1104#issuecomment-388651406) with that package version not being in one definitive location, but in two .nuspec dependency lists and that targets file.

The various cases are 
`RoslynCodeTaskFactory` is used by the consuming project -- then `$(RoslynCodeTaskFactory)` will be set, and whichever version that is will be used directly 
`RoslynCodeTaskFactory` is available as a consequence of being an Eto dependency as per this PR -- then while the .nuspec says 2.0.7 we should be good.
`RoslynCodeTaskFactory` is available on the machine but only in some other version -- well, the build will fail with an advisory message, and with the latest change, `$(RoslynCodeTaskFactoryVersion)` can be set in the consuming project to pick up what's available.

**Comment by cwensley** [comment]
Hey @SteveGilham, thanks for the insight.  I just looked at RoslynCodeTaskFactory, and apparently it already should be including the props file if it's added as a package reference.  

This means those lines that manually import `Sdk.props` from the RoslynCodeTaskFactory package should not even be necessary.

You've already added the dependency to Eto's nuspec so that should take care of it, correct?

**Comment by SteveGilham** [comment]
If the package is included in the project, then the conditional manual inclusion isn't necessary.

Without it, existing projects that don't include RoslynCodeTaskFactory would break, though.  The manual inclusion is purely there to provide backwards compatibility (the package just has to exist on the machine as a dependency of the Eto package for existing builds to just keep on working).  If you're happy to make a breaking release, then that can go.

**Comment by cwensley** [comment]
Thanks, I appreciate that you're thinking about backwards compatibility.  However, I don't think I'd be a breaking change - updating to the new Eto.Mac/Eto.Mac64 nuget packages will automatically pull in the RoslynCodeTaskFactory.

It'd only break when MacTemplate.targets is imported manually, which in that case I don't care for backwards compatibility as it isn't a supported way to use it.  Even so, adding the dependency to your project would fix it which the error message would point them to do.  

I'd rather not have the hack and just keep things as straightforward as possible to ease maintenance going forward (e.g. RoslynCodeTaskFactory could change how it is packaged).

**Comment by cwensley** [comment]
Excellent!  Thanks for the contribution!

**Pull request merged**